tests/samples_test.py

```diff
@@ -0,0 +1,405 @@
+            numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+        ],
+    ),
+    (smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 0), [0]),
+    (smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 1), [0, -1, 1]),
+    (
+        smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 2),
+        [0, -1.0, 1.0, -1 / numpy.sqrt(2), 1 / numpy.sqrt(2)],
+    ),
+    (
+        smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 3),
+        [
+            0,
+            -1.0,
+            1.0,
+            -1 / numpy.sqrt(2),
+            1 / numpy.sqrt(2),
+            -numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+            -numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+        ],
+    ),
+    (
+        smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 4),
+        [
+            0,
+            -1.0,
+            1.0,
+            -1 / numpy.sqrt(2),
+            1 / numpy.sqrt(2),
+            -numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+            -numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+        ],
+    ),
+    (
+        smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 5),
+        [
+            0,
+            -1.0,
+            1.0,
+            -1 / numpy.sqrt(2),
+            1 / numpy.sqrt(2),
+            -numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+            -numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+            -numpy.sqrt(2 + numpy.sqrt(2 + numpy.sqrt(2))) / 2,
+            -numpy.sqrt(2 + numpy.sqrt(2 - numpy.sqrt(2))) / 2,
+            -numpy.sqrt(2 - numpy.sqrt(2 - numpy.sqrt(2))) / 2,
+            -numpy.sqrt(2 - numpy.sqrt(2 + numpy.sqrt(2))) / 2,
+            numpy.sqrt(2 - numpy.sqrt(2 + numpy.sqrt(2))) / 2,
+            numpy.sqrt(2 - numpy.sqrt(2 - numpy.sqrt(2))) / 2,
+            numpy.sqrt(2 + numpy.sqrt(2 - numpy.sqrt(2))) / 2,
+            numpy.sqrt(2 + numpy.sqrt(2 + numpy.sqrt(2))) / 2,
+        ],
+    ),
+    (smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 0), [0]),
+    (
+        smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 1),
+        [0, 2 * numpy.pi / 3, 4 * numpy.pi / 3],
+    ),
+    (
+        smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 2),
+        2
+        * numpy.pi
+        * numpy.array(
+            [
+                0,
+                1 / 3,
+                2 / 3,
+                1 / 9,
+                2 / 9,
+                4 / 9,
+                5 / 9,
+                7 / 9,
+                8 / 9,
+            ]
+        ),
+    ),
+]
+
+sample_points_ids = [
+    "Non-nested Cheb [0]",
```

Comment:
I would name these to match the point sets we have and don't include "non-nested", like
```suggestion
    "ClenshawCurtis [0]",
```